distribution/docker/src/docker/Dockerfile

```diff
@@ -10,70 +10,174 @@
 
   https://docs.groovy-lang.org/latest/html/api/groovy/text/SimpleTemplateEngine.html
 */ %>
+
 ################################################################################
-# Build stage 0 `builder`:
-# Extract elasticsearch artifact
-# Set gid=0 and make group perms==owner perms
+# Step 1. Build curl statically. Installing it from RPM on CentOS pulls in too
+# many dependencies.
 ################################################################################
+FROM alpine:latest AS curl
 
-FROM centos:7 AS builder
+ENV VERSION 7.71.0
+ENV TARBALL_URL https://curl.haxx.se/download/curl-\${VERSION}.tar.xz
+ENV TARBALL_PATH curl-\${VERSION}.tar.xz
+
+# Install dependencies
+RUN apk add gnupg gcc make musl-dev openssl-dev openssl-libs-static file
+
+RUN mkdir /work
+WORKDIR /work
+
+# Fetch curl sources and files for validation
+RUN wget "https://daniel.haxx.se/mykey.asc" -O "curl-gpg.pub" && \\
+    wget "\${TARBALL_URL}.asc" -O "\${TARBALL_PATH}.asc" && \\
+    wget "\${TARBALL_URL}" -O "\${TARBALL_PATH}"
+
+# Validate source
+RUN gpg --import --always-trust "curl-gpg.pub" && \\
+    gpg --verify "\${TARBALL_PATH}.asc" "\${TARBALL_PATH}"
+
+# Unpack and build
+RUN tar xfJ "\${TARBALL_PATH}" && \\
+    cd "curl-\${VERSION}" && \\
+    ./configure --disable-shared --with-ca-fallback --with-ca-bundle=/etc/pki/tls/certs/ca-bundle.crt && \\
+    make curl_LDFLAGS="-all-static" && \\
+    cp src/curl /work/curl && \\
+    strip /work/curl
+
+################################################################################
+# Step 2. Create a minimal root filesystem directory. This will form the basis
+# for our image.
+################################################################################
+FROM centos:7 AS rootfs
+
+ENV BUSYBOX_VERSION 1.31.0
+ENV TINI_VERSION 0.19.0
+
+# Start off with an up-to-date system
+RUN yum update --setopt=tsflags=nodocs -y
+
+# Create a directory into which we will install files
+RUN mkdir /rootfs
+
+# Create required devices
+RUN mkdir -m 755 /rootfs/dev && \\
+    mknod -m 600 /rootfs/dev/console c 5 1 && \\
+    mknod -m 600 /rootfs/dev/initctl p && \\
+    mknod -m 666 /rootfs/dev/full c 1 7 && \\
+    mknod -m 666 /rootfs/dev/null c 1 3 && \\
+    mknod -m 666 /rootfs/dev/ptmx c 5 2 && \\
+    mknod -m 666 /rootfs/dev/random c 1 8 && \\
+    mknod -m 666 /rootfs/dev/tty c 5 0 && \\
+    mknod -m 666 /rootfs/dev/tty0 c 4 0 && \\
+    mknod -m 666 /rootfs/dev/urandom c 1 9 && \\
+    mknod -m 666 /rootfs/dev/zero c 1 5
+
+# Install a minimal set of dependencies, and some for Elasticsearch
+RUN yum --installroot=/rootfs --releasever=/ --setopt=tsflags=nodocs \\
+      --setopt=group_package_types=mandatory -y  \\
+      --skip-broken \\
+      install basesystem bash zip zlib
```

Comment:
How about adding `pigz`? 
It's very useful for compressing large heap dumps. It's not a must-have from Cloud's perspective (we can fall back to `gzip`) but since we will be lacking package manager it'd make a lot of sense to add it here.

Reply:
@mieciu It's my assumption that heap dumps would be written to a volume that is mounted on the host (otherwise, they vanish when the container dies, which will happen when the process died because of an OOME), and thus could be processed on the host external to the container.

Reply:
That's a good point - @mieciu what is Cloud doing with heapdumps exactly - compressing them in the container?

Reply:
We could move the compression out of the ES container and into the allocator container. Then pigz is always available (if we want it to be).

Reply:
So I'm happy to do something to smooth the way, though compressing heapdumps does feel like an orchestration concern, so ultimately I think the allocator should handle it in whatever way it deems fit.